### ISSUE TITLE
False positive IDE0060 for partial method with SG

### ISSUE THREAD

**Issue by huoyaoyuan** [dotnet/roslyn#57814]
**Version Used**: 
VS 17.0.0

**Steps to Reproduce**:

1. Write a source generator to define a partial method with parameter for a class
2. Implement the partial method in user code, without touching the parameter

**Expected Behavior**:
No IDE0060 reported, at least when the source generator is fully loaded.

**Actual Behavior**:
IDE0060 unused parameter reported for the partial method.

Should we disable IDE0060 for any partial method? Their signature should match for different sources of code, and user code may not change it.

**Comment by kaleidocore** [comment]
As a workaround you can use the `SuppressMessageAttribute` in the generated file:

```
		[System.Diagnostics.CodeAnalysis.SuppressMessage("Style", "IDE0060: Remove unused parameter")]
		partial void MyPartialMethod(int someParam);
```
